drivers/gpio/gpio_mchp_xec_v2.c

```diff
@@ -81,103 +82,118 @@ static inline void xec_mask_write32(uintptr_t addr, uint32_t mask, uint32_t val)
+		return -EINVAL;
 	}
 
-	/* Push-pull or open drain */
-	mask |= MCHP_GPIO_CTRL_BUFT_MASK;
+	int ret = gpio_xec_validate_flags(flags);
 
-	if ((flags & GPIO_OPEN_DRAIN) != 0U) {
-		/* Open drain */
-		pcr1 |= MCHP_GPIO_CTRL_BUFT_OPENDRAIN;
-	} else {
-		/* Push-pull */
-		pcr1 |= MCHP_GPIO_CTRL_BUFT_PUSHPULL;
+	if (ret) {
+		return ret;
 	}
 
-	/* Use GPIO output register to control pin output, instead of
-	 * using the control register (=> alternate output disable).
-	 */
-	mask |= MCHP_GPIO_CTRL_AOD_MASK;
-	pcr1 |= MCHP_GPIO_CTRL_AOD_DIS;
+	pcr1_addr = pin_ctrl_addr(dev, pin);
+	pcr1 = sys_read32(pcr1_addr);
 
-	/* Make sure disconnected on first control register write */
 	if (flags == GPIO_DISCONNECTED) {
-		pcr1 |= MCHP_GPIO_CTRL_PWRG_OFF;
+		pcr1 = (pcr1 & ~MCHP_GPIO_CTRL_PWRG_MASK) | MCHP_GPIO_CTRL_PWRG_OFF;
+		sys_write32(pcr1, pcr1_addr);
+		return 0;
 	}
 
-	/* Now write contents of pcr1 variable to the PCR1 register that
-	 * corresponds to the GPIO being configured.
-	 * AOD is 1 and direction is input. HW will allow use to set the
-	 * GPIO parallel output bit for this pin and with the pin direction
-	 * as input no glitch will occur.
-	 */
-	xec_mask_write32(pcr1_addr, mask, pcr1);
+	/* final pin state will be powered */
+	pcr1_new = MCHP_GPIO_CTRL_PWRG_VTR_IO;
 
-	if ((flags & GPIO_OUTPUT) != 0U) {
-		if ((flags & GPIO_OUTPUT_INIT_HIGH) != 0U) {
-			sys_set_bit(pout_addr, pin);
-		} else if ((flags & GPIO_OUTPUT_INIT_LOW) != 0U) {
-			sys_clear_bit(pout_addr, pin);
+	/* always enable input pad */
+	if (pcr1 & BIT(MCHP_GPIO_CTRL_INPAD_DIS_POS)) {
+		pcr1 &= ~BIT(MCHP_GPIO_CTRL_INPAD_DIS_POS);
+		sys_write32(pcr1, pcr1_addr);
+	}
+
+	if (flags & GPIO_OUTPUT) {
+		pcr1_new |= BIT(MCHP_GPIO_CTRL_DIR_POS);
+		msk |= BIT(MCHP_GPIO_CTRL_OUTVAL_POS);
+		if (flags & GPIO_OUTPUT_INIT_HIGH) {
+			pcr1_new |= BIT(MCHP_GPIO_CTRL_OUTVAL_POS);
+		} else if (flags & GPIO_OUTPUT_INIT_LOW) {
+			pcr1_new &= ~BIT(MCHP_GPIO_CTRL_OUTVAL_POS);
+		} else { /* copy current input state to output state */
+			if ((pcr1 & MCHP_GPIO_CTRL_PWRG_MASK) == MCHP_GPIO_CTRL_PWRG_OFF) {
+				pcr1 &= ~(MCHP_GPIO_CTRL_PWRG_MASK);
+				pcr1 |= MCHP_GPIO_CTRL_PWRG_VTR_IO;
+				sys_write32(pcr1, pcr1_addr);
+			}
+			pcr1 = sys_read32(pcr1_addr);
+			if (pcr1 & BIT(MCHP_GPIO_CTRL_INPAD_VAL_POS)) {
+				pcr1_new |= BIT(MCHP_GPIO_CTRL_OUTVAL_POS);
+			} else {
+				pcr1_new &= ~BIT(MCHP_GPIO_CTRL_OUTVAL_POS);
+			}
 		}
+		if (flags & GPIO_LINE_OPEN_DRAIN) {
+			pcr1_new |= BIT(MCHP_GPIO_CTRL_BUFT_POS);
+		}
+	}
 
-		mask = MCHP_GPIO_CTRL_DIR_MASK;
-		pcr1 = MCHP_GPIO_CTRL_DIR_OUTPUT;
-		xec_mask_write32(pcr1_addr, mask, pcr1);
+	if (flags & (GPIO_PULL_UP | GPIO_PULL_DOWN)) {
+		msk |= MCHP_GPIO_CTRL_PUD_MASK;
+		/* both bits specifies repeater mode */
+		if (flags & GPIO_PULL_UP) {
+			pcr1_new |= MCHP_GPIO_CTRL_PUD_PU;
+		}
+		if (flags & GPIO_PULL_DOWN) {
+			pcr1_new |= MCHP_GPIO_CTRL_PUD_PD;
+		}
 	}
 
+	/* Problem, if pin was power gated off we can't read input.
```

Comment:
Conventionally, block comments in Zephyr usually keep `/*` and `*/` on a separate lines.

https://docs.zephyrproject.org/latest/contribute/guidelines.html#coding-style
https://kernel.org/doc/html/latest/process/coding-style.html#commenting

Reply:
Updated the block comments as per coding-style.